src/wccp2.cc

```diff
@@ -519,8 +519,7 @@ wccp2_add_service_list(int service, int service_id, int service_priority,
     wccp2_update_service(wccp2_service_list_ptr, service, service_id,
                          service_priority, service_proto, service_flags, ports);
     wccp2_service_list_ptr->wccp2_security_type = security_type;
-    memset(wccp2_service_list_ptr->wccp_password, 0, WCCP2_PASSWORD_LEN + 1);
-    strncpy(wccp2_service_list_ptr->wccp_password, password, WCCP2_PASSWORD_LEN);
+    xstrncpy(wccp2_service_list_ptr->wccp_password, password, WCCP2_PASSWORD_LEN);
```

Comment:
This will truncate 1 character off the end of passwords that fill the buffer because the n parameter to xstrncpy() includes the c-string terminator character.

For these conversions you need to use the length parameter from the memset() as buffer size instead of the pre-shortened one from the old strncpy().

Reply:
You're correct, thanks.  Fixed.